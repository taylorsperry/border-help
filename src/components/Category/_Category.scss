@import '../../variables';

.category {
  width: 80%;
  border: solid $navy 1px;
}