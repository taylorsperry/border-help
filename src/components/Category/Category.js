import React, { Component } from 'react'
import Item from '../Item/Item'
import './_Category.scss'

export class Category extends Component {
 render() {
  const { catName, data, callFetch } = this.props
  if(!data.length) {
    callFetch()
  }
  const displayItems = data.map(item => (
    <Item key={item.id} item={item} />
    ))
  return (
    <div className='category'>
      <h2 className='cat-name'>{catName}</h2>
      {displayItems}
    </div>
  )
}
}

export default Category