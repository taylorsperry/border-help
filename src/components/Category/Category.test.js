import React from 'react'
import { Category } from './Category'
import { shallow } from 'enzyme'

describe('Category', () => {
  let wrapper
  it('should match the snapshot', () => {
    let mockProps = {
      catName: 'name',
      data: ['one', 'two'],
      callFetch: jest.fn()
    }
    wrapper = shallow(<Category {...mockProps}/>)
    expect(wrapper).toMatchSnapshot()
  })

  it('should callFetch if !data.length', () => {
    let mockProps = {
      catName: 'name',
      data: [],
      callFetch: jest.fn()
    }
    wrapper = shallow(<Category {...mockProps}/>)
    expect(wrapper.instance().props.callFetch).toHaveBeenCalled()
  })
})