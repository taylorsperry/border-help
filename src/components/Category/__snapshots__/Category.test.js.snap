// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category should match the snapshot 1`] = `
<div
  className="category"
>
  <h2
    className="cat-name"
  >
    name
  </h2>
  <Item
    item="one"
  />
  <Item
    item="two"
  />
</div>
`;
